decisions/migrations.md

```diff
@@ -0,0 +1,262 @@
+# Migrations
+
+- *Date*: 2022-10-19
+- *Deciders*:
+- *Status*: DRAFT
+- *Outcome*: 
+- *Related Material*: [Database migration Issue](https://github.com/openmsupply/open-msupply/issues/704)
+
+## Requirements
+
+1. Update schema without re-initialising database
+2. Migrate data
+   a. Before and/or after schema update
+   b. Without schema update (just data migration)
+3. Do above only for the migrations that are applicable to from current database version to current app version (don't re-run old migrations)
+4. Record migration results
+5. Easy way to explore full schema (all of the migrations applied, for a particular version)
+6. Test data migrations
+7. Should work with RC versions
+8. (maybe) Consolidate common sql for PG/Sqlite (we have quite a few duplicate sql statements)
+9. (maybe) Use strongly typed diesel db if possible, for data migrations
+10. Migration code needs to run in isolation (no production code should run in parallel)
```

Comment:
yes. we also have an issue with the potential of long-running data migrations delaying the server start up; which has happened with mSupply and resulted in the exploring of UI to indicate when migration is running / completed.

I wonder about treating long-running data migration as a separate concern; in which we allow quick migration as part of the usual process, but have another option for long running tasks; something that takes the system offline, runs migration with some kind of progress indication and then gets it back online again after validity checks. Wonder if the server could serve up a static `site under maintenance` page while running migration.. could be a cli method, or an async server method with a status / progress endpoint..

Reply:
👍 for both progress indication and site under maintenance, added to epic https://github.com/openmsupply/open-msupply/issues/704